plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/StmtConversionVisitor.kt

```diff
@@ -298,14 +312,49 @@ object StmtConversionVisitor : FirVisitor<Exp, StmtConversionContext<ResultTrack
         variableAssignment: FirVariableAssignment,
         data: StmtConversionContext<ResultTrackingContext>,
     ): Exp {
+
         // It is not entirely clear whether we can get away with ignoring the distinction between
         // lvalues and rvalues, but let's try to at first, and we'll fix it later if it turns out
         // not to work.
-        val convertedLValue = data.convert(variableAssignment.lValue)
         val lValueType = data.embedType(variableAssignment.lValue)
         val convertedRValue = data.convert(variableAssignment.rValue)
         val rValueType = data.embedType(variableAssignment.rValue)
-        data.addStatement(Stmt.assign(convertedLValue, convertedRValue.convertType(rValueType, lValueType)))
+
+        if (variableAssignment.lValue.isAssigningToClassField) {
+
+            val lValue = variableAssignment.lValue as FirPropertyAccessExpression
+            val lValueSymbol = lValue.calleeSymbol as FirPropertySymbol
+            val receiver = data.convert(lValue.dispatchReceiver!!)
+            /*
+            According to the setter symbol, we should invoke a method, or assigning a value.
+
+            Setters are only relevant for variable properties.
+            Compared to the getter's, the setter symbols require more information about the backing field.
+            If a custom setter is defined on a variable property. this also exposes a backing field.
+            To check the existence of a custom setter, we can look for a function body in a `setterSymbol`.
```

Comment:
I'm struggling a bit to understand this.  When you say `variable properties`, what kind of properties don't you include there?  A single overview with what kinds of getters and setters there are and how they are encoded in the FIR would probably work better here.

It helps to write this in a declarative, not procedural style: it's not about what _we_ should do, but about the rules that govern this data structure.